virttest/vt_cluster/node_properties.py

```
@@ -0,0 +1,88 @@
+    :raises: OSError if file read operations fail
+    """
+    if not os.path.exists(cluster.metadata_file):
+        LOG.warning(f"Metadata file {cluster.metadata_file} does not exist")
+        return {}
+
+    try:
+        with open(cluster.metadata_file, "r") as metadata_file:
+            props = json.load(metadata_file)
+            return props
+    except json.JSONDecodeError as e:
+        LOG.error(f"Failed to parse metadata file {cluster.metadata_file}: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)